ruby/example_code/elasticbeanstalk/spec/list_apps_spec.rb

```diff
@@ -0,0 +1,13 @@
+require "rspec"
+require_relative "../list_apps"
+
+describe ElasticBeanstalkManager do
+  let(:eb_client) { Aws::ElasticBeanstalk::Client.new }
```

Comment:
in rspecs we typically use `stub_responses: true`.  This prevents the specs from making actual network calls (and therfore requiring credentials, ect). 

You can also actually stub expected data which makes writting tests much easier! This is heavily used in the SDK itself.  See: https://aws.amazon.com/blogs/developer/advanced-client-stubbing-in-the-aws-sdk-for-ruby-version-3/

I'm not sure if these specs are intended to be part of examples shown to users or not - I do think its very useful to show users examples of using client stubbing in their own tests.

Reply:
We now have an automated integration test runner that builds all the examples into a Dockerfile and runs them with credentials in AWS. We don't tend to treat tests as example code itself, though I'm not against this in principle; we just don't have the manpower to make sure they all work correctly end to end in AWS and also offer examples for stubbing.